compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -1132,9 +1132,15 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         &mut self,
         op: rustc_codegen_ssa::common::AtomicRmwBinOp,
         dst: &'ll Value,
-        src: &'ll Value,
+        mut src: &'ll Value,
         order: rustc_codegen_ssa::common::AtomicOrdering,
     ) -> &'ll Value {
+        // The only RMW operation that LLVM supports on pointers is compare-exchange.
```

Comment:
should we report this somewhere upstream? Or at least track it with a FIXME or issue?

Reply:
Which operations does Rust use on pointers? To the most part these just don't make sense, at least not in a straightforward way (even something like `add` would run up against the provenance question).